test/ServiceRegistry.js

```diff
@@ -1569,6 +1569,163 @@ describe("ServiceRegistry", function () {
+                // Getting a real multisig address
+                const multisig = await ethers.getContractAt("GnosisSafe", proxyAddress);
+
+                // Terminate a service after some time since there's a need to add agent instances
+                await serviceRegistry.connect(serviceManager).terminate(serviceOwnerAddress, serviceId);
+
+                // Unbond the agent instance in order to update the service
+                await serviceRegistry.connect(serviceManager).unbond(operator, serviceId);
+
+                const safeContracts = require("@gnosis.pm/safe-contracts");
+                // At this point of time the agent instance gives the ownership rights to the service owner
+                // In other words, swap the owner of the multisig to the service owner (agent instance to give up rights for the service owner)
+                // Since there was only one agent instance, the previous multisig owner address is the sentinel one defined by gnosis (0x1)
+                const sentinelOwners = "0x" + "0".repeat(39) + "1";
+                let nonce = await multisig.nonce();
+                const txHashData = await safeContracts.buildContractCall(multisig, "swapOwner",
+                    [sentinelOwners, agentInstances[0].address, serviceOwnerAddress], nonce, 0, 0);
+                const signMessageData = await safeContracts.safeSignMessage(agentInstances[0], multisig, txHashData, 0);
+                await safeContracts.executeTx(multisig, txHashData, [signMessageData], 0);
+
+                // Updating a service
+                await serviceRegistry.connect(serviceManager).update(serviceOwnerAddress, configHash, [1], [[4, regBond]],
+                    newMaxThreshold, serviceId);
+
+                // Activate agent instance registration
+                await serviceRegistry.connect(serviceManager).activateRegistration(serviceOwnerAddress, serviceId, {value: regDeposit});
+
+                /// Register agent instance
+                await serviceRegistry.connect(serviceManager).registerAgents(operator, serviceId,
+                    [agentInstances[0].address, agentInstances[1].address, agentInstances[2].address, agentInstances[3].address],
+                    [agentId, agentId, agentId, agentId], {value: 4 * regBond});
+
+                // Change the existent multisig owners and threshold in a multisend transaction using the service owner access
+                let callData = [];
+                let txs = [];
+                nonce = await multisig.nonce();
+                // Add the addresses skipping the first one, but keep the threshold the same
+                for (let i = 0; i < agentInstances.length - 1; i++) {
+                    callData[i] = multisig.interface.encodeFunctionData("addOwnerWithThreshold", [agentInstances[i+1].address, 1]);
+                    txs[i] = safeContracts.buildSafeTransaction({to: multisig.address, data: callData[i], nonce: 0});
+                }
```

Comment:
We add owners starting from the second one, as the first one is going to be swapped with the current service owner being the multisig owner.